service/controller/v3/debugger/debugger.go

```diff
@@ -0,0 +1,45 @@
+package debugger
+
+import (
+	"context"
+
+	"github.com/Azure/azure-sdk-for-go/services/resources/mgmt/2018-02-01/resources"
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+
+	"github.com/giantswarm/azure-operator/service/controller/v3/key"
+)
+
+type Config struct {
+	Logger micrologger.Logger
+}
+
+type Debugger struct {
```

Comment:
We introduce a new concept here so we can configure a debugger where it is needed so we can log more information we want to perceive in certain situations based on conditions that might be more complex.